vunit/ui/__init__.py

```diff
@@ -489,6 +492,26 @@ def add_compile_option(self, name: str, value: str, allow_empty: Optional[bool]
         for source_file in check_not_empty(source_files, allow_empty, "No source files found"):
             source_file.add_compile_option(name, value)
 
+
+    def set_global_compile_options(self, name: str, value: str, allow_empty: Optional[bool] = False):
```

Comment:
Is this a way to add compile options to future added source files? Is that necessary for Xcelium support or a separate feature?

Reply:
This was needed to achieve acceptable performance for Xcelium via single file compilation (We saw compilation times go from ~2m to ~11s on our NFS share). The other option was adding all compilation options into a hash map and then de-duping them, but in early testing that proved to be very error prone based on how users might decide to compose command line options in Python.